astropy/io/ascii/src/tokenizer.h

```diff
@@ -13,6 +13,14 @@
 
 #ifdef _MSC_VER
 #define inline __inline
+    #ifndef NAN
+        static const unsigned long __nan[2] = {0xffffffff, 0x7fffffff};
```

Comment:
I think this counts on `long` being 32-bits, which isn't true on all platforms.  May include `stdint.h` and use `uint32_t`?